### PR TITLE
Bokeh 2 support

### PR DESCRIPTION
**Pull request by AjayThorve**
- updated dependencies: bokeh 2.1.1, panel 0.9.*
- added support for panel.pane.DeckGL
- removed custom deckgl extension
- added button to open dashboard

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cuxfilter/charts/deckgl/deckgl.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def choropleth(
 
     geo_color_palette: bokeh.palette,  default bokeh.palettes.Inferno256
 
-    nan_color: str, default white
+    nan_color: hex color code, default '#d3d3d3'
```
Comment (bryevdv) [inline python/cuxfilter/charts/deckgl/deckgl.py:73]: Seen this hex code a few places now, maybe make it a module level `CUXF_NAN_COLOR` somewhere convenient?

File: python/cuxfilter/dashboard.py [hunk 2]
```diff
@@ -27,7 +29,7 @@
 
 
 def _create_dashboard_url(notebook_url: str, port: int):
-    return f"http://{notebook_url}/proxy/{port}/"
+    return f"http://{notebook_url}/proxy/{port}/ "
```
Comment (bryevdv) [inline python/cuxfilter/dashboard.py:32]: Is the trailing space needed for some reason?

File: python/cuxfilter/charts/deckgl/bindings/panel_deck.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    width = param.Integer(400)
+
+    def get_tooltip_html(self):
+        html_str = ""
+        tooltip_columns = list(
+            set(self.data.columns)
+            - set(["index", "coordinates"] + list(self.colors.columns))
+        )
+        for i in tooltip_columns:
+            html_str += "<b> %s </b>: {%s} <br><br>" % (i, i)
```
Comment (bryevdv) [inline python/cuxfilter/charts/deckgl/bindings/panel_deck.py:44]: could use f-string here and avoid old-style positional substitution (tho braces would need escaping so maybe it's a wash)

File: python/cuxfilter/charts/deckgl/plots.py [hunk 4]
```diff
@@ -48,6 +50,31 @@ class Choropleth(BaseChoropleth):
...
+            for i, b in enumerate(BREAKS):
+                if val < b:
+                    return list(
+                        ImageColor.getrgb(self.geo_color_palette[i])
+                    ) + [255]
+            return list(ImageColor.getrgb(self.geo_color_palette[i])) + [255]
+
+        self.source_df[self.rgba_columns] = pd.DataFrame(
+            self.source_df[self.color_column].apply(color_scale).tolist()
+        )
```
Comment (bryevdv) [inline python/cuxfilter/charts/deckgl/plots.py:76]: I feel like you [`pandas.cut`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.cut.html) could be used here to efficiently get the palette indices and then the colors could be computed with a list comprehension (untested sketch)
```
nan_color = list(ImageColor.getrgb(self.nan_color)) + [50]

x = self.source_df[self.color_column]
inds = pd.cut(x, BREAKS, labels=False, include_lowest=True)

colors = [nan_color if np.isnan(i) else list(ImageColor.getrgb(self.geo_color_palette[i])) + [255] for i in inds]
```

**Comment by bryevdv** [comment]
LGTM (modulo ongoing packaging concerns) just a few minor comments

File: python/cuxfilter/charts/deckgl/plots.py (hunk 4 above)
Comment (AjayThorve) [reply python/cuxfilter/charts/deckgl/plots.py:76]: Yes, the proposed method is actually around 5-6x faster (1000 rows)

File: python/cuxfilter/dashboard.py (hunk 2 above)
Comment (AjayThorve) [reply python/cuxfilter/dashboard.py]: removed

File: python/cuxfilter/charts/deckgl/bindings/panel_deck.py (hunk 3 above)
Comment (AjayThorve) [reply python/cuxfilter/charts/deckgl/bindings/panel_deck.py]: using f-string now

File: python/cuxfilter/charts/deckgl/deckgl.py (hunk 1 above)
Comment (AjayThorve) [reply python/cuxfilter/charts/deckgl/deckgl.py]: added CUXF_NAN_COLOR as a module level constant in cuxfilter.charts.constants

File: python/cuxfilter/dashboard.py [hunk 5]
```diff
@@ -520,6 +520,13 @@ def show(
                 **kwargs,
             )
         self._current_server_type = "show"
+        b = pn.widgets.Button(
+            name="open cuxfilter dashboard", button_type="success"
+        )
+        b.js_on_click(
+            args={"target": dashboard_url}, code="window.open(target)"
+        )
+        return pn.Row(b)
```
Comment (bryevdv) [inline python/cuxfilter/dashboard.py:529]: 👍

**Comment by AjayThorve** [comment]
rerun tests

**Comment by AjayThorve** [comment]
Rerun tests

**Pull request merged**